### PR TITLE
fix(typescript): workaround for trouble with aggregate-error type exports (#245)

### LINKED ISSUE #245: [TypeScript] v7.11.2 has a hard requirement on specific TS configs
```ts
@octokit/webhooks/dist-types/types.d.ts(1,13): error TS1259: Module
'"/home/robcresswell/code/new-proj/node_modules/aggregate-error/index"'
can only be default-imported using the 'allowSyntheticDefaultImports' flag
```

This seems to have recently started popping up; adding `skipLibCheck: true` makes it go away :( 

I can push a minimal repro in a couple days if that helps too

### PR DESCRIPTION
**Pull request by gr2m**
resolves #245
resolves #270

 thanks @andrewbranch /cc @dominguezcelada

### CODE REVIEW

File: src/middleware/middleware.ts [hunk 1]
```diff
@@ -6,6 +6,8 @@ import { debug } from "debug";
 import { IncomingMessage, ServerResponse } from "http";
 import { State, WebhookError, WebhookEventHandlerError } from "../types";
 import { EventNames } from "../generated/event-names";
+
+// @ts-ignore to address #245
 import AggregateError from "aggregate-error";
```
Comment (andrewbranch) [inline src/middleware/middleware.ts]: Is this import actually used in this file?

**Review by andrewbranch — DISMISSED** [review]
You’ll want to try a pika build and check that the dist output looks like you expect, but I think this should work.

**Comment by gr2m** [comment]
> You’ll want to try a pika build and check that the dist output looks like you expect, but I think this should work.

I use `pika-ci` to generate previews of the to-be-published assets. It can be downloaded here:
https://github.pika.dev/octokit/webhooks.js/pr/270. It all looks good as far as I can tell?

File: src/middleware/middleware.ts (hunk 1 above)
Comment (gr2m) [reply src/middleware/middleware.ts]: huh, good catch :)

Comment (gr2m) [reply src/middleware/middleware.ts]: resolved via 034ae6e

**Comment by gr2m** [comment]
@dominguezcelada are you happy with these changes?

@robcresswell can you please check if that resolves https://github.com/octokit/webhooks.js/issues/245 for you? To test this PR, you can `npm install https://github.pika.dev/octokit/webhooks.js/pr/270`

**Comment by andrewbranch** [comment]
Tested by throwing that bundle into my node_modules. Looks good 🌟

**Comment by oscard0m** [comment]
Tested in a local repo I was playing with. Running `tsc` succeeds now.

I created a new issue to make sure we do a followup on this (since the existing one will be closed when we merge this): https://github.com/octokit/webhooks.js/issues/271

File: src/types.ts [hunk 2]
```diff
@@ -47,3 +46,12 @@ export interface WebhookEventHandlerError extends AggregateError<WebhookError> {
    */
   errors: WebhookError[];
 }
+
+declare class AggregateError<T extends Error = Error>
```
Comment (oscard0m) [inline src/types.ts:50]: Even though is a temporary fix, we should mention and give credit to the original repo? https://github.com/octokit/webhooks.js/pull/248#issuecomment-695092969 @gr2m

**Review by oscard0m — COMMENTED** [review]
Looks good to me. Only check small comment

File: src/types.ts (hunk 2 above)
Comment (gr2m) [reply src/types.ts:50]: oopsies sorry I missed your comment. You are right. I'll send a follow up pR

**Pull request merged**